src/Core/cldj_sub_mod.F90

```diff
@@ -69,34 +92,50 @@ SUBROUTINE CLOUD_JX (U0,SZA,RFL,SOLF,LPRTJ,PPP,ZZZ,TTT,HHH,DDD,  &
+      integer, intent(in)                    :: IRAN
+      real*8,  intent(in)                    :: U0     !v7.7
+      real*8,  intent(in)                    :: SZA    !v7.7
+      real*8,  intent(in)                    :: SOLF   !v7.7
+      real*8,  intent(inout)                 :: CLDCOR !v7.7
+      real*8,intent(in), dimension(5,W_+W_r) :: RFL
       logical, intent(in)                    :: LPRTJ
-      real*8,  intent(in), dimension(L1U+1)  :: PPP,ZZZ
-      real*8,  intent(in), dimension(L1U  )  :: TTT,HHH,DDD,RRR,OOO,CCC
-      real*8,  intent(in), dimension(L1U  )  :: LWP,IWP,REFFL,REFFI
+      real*8,  intent(in), dimension(L1U+1)  :: PPP
+      real*8,  intent(in), dimension(L1U+1)  :: ZZZ
+      real*8,  intent(in), dimension(L1U  )  :: TTT
+      real*8,  intent(in), dimension(L1U  )  :: HHH
+      real*8,  intent(in), dimension(L1U  )  :: DDD
+      real*8,  intent(in), dimension(L1U  )  :: RRR
+      real*8,  intent(in), dimension(L1U  )  :: OOO
+      real*8,  intent(in), dimension(L1U  )  :: CCC
+      real*8,  intent(in), dimension(L1U  )  :: LWP
+      real*8,  intent(in), dimension(L1U  )  :: IWP
+      real*8,  intent(in), dimension(L1U  )  :: REFFL
+      real*8,  intent(in), dimension(L1U  )  :: REFFI
       real*8,  intent(in), dimension(L1U,ANU):: AERSP
       integer, intent(in), dimension(L1U,ANU):: NDXAER
       real*8,  intent(in), dimension(L1U  )  :: CLDF
       integer, intent(in), dimension(L1U  )  :: CLDIW
 ! reports out the JX J-values, upper level program converts to CTM chemistry J's
-      real*8,  intent(out), dimension(L1U-1,NJXU):: VALJXX
-      real*8,  intent(out), dimension(S_+2,L1U)  :: SKPERD
-      real*8,  intent(out), dimension(6)         :: SWMSQ
-      real*8,  intent(out), dimension(L1U)       :: OD18
-      real*8,  intent(out), dimension(NQD_)      :: WTQCA
-      integer, intent(out)                       :: NICA,JCOUNT
-      logical, intent(out)                       :: LDARK
+      real*8,intent(out), dimension(L1U-1,NJXU):: VALJXX
+      real*8,  intent(out), dimension(S_+2,L1U):: SKPERD
+      real*8,  intent(out), dimension(6)       :: SWMSQ
+      real*8,  intent(out), dimension(L1U)     :: OD18
+      real*8,  intent(out), dimension(NQD_)    :: WTQCA
+      integer, intent(out)                     :: NICA
+      integer, intent(out)                     :: JCOUNT
+      logical, intent(out)                     :: LDARK
 !-----------------------------------------------------------------------
       logical  LPRTJ0
       integer  I,II,J,K,L,M,N, LTOP, NRG,IRANX
       real*8   CLDFR, XRAN, FSCALE, QCAOD, WTRAN
       real*8,  dimension(L1U)     :: LWPX,IWPX,REFFLX,REFFIX
       real*8,  dimension(LWEPAR)  :: CLTL,CLTI, CLT,CLDX
       integer, dimension(LWEPAR)  :: NCLDF
-      integer, dimension(9)       :: GBOT,GTOP,GLVL,GNR,GCMX  ! # max-verlap groups set at 9
+      ! # max-verlap groups set at 9
+      integer, dimension(9)       :: GBOT,GTOP,GLVL,GNR,GCMX

```

Comment:
See code below.  The loop over L isn't necessary to initialize the arrays.  The code could be:
```F90
      LPRTJ0  = LPRTJ
      JCOUNT  = 0
      NICA    = 0
      LWPX    = 0.d0
      IWPX    = 0.d0
      REFFLX  = 0.d0
      REFFIX  = 0.d0
      VALJXX  = 0.d0    ! zero J's Heating R's in case LDARK is returned
      SKPERD  = 0.d0
      SWMSQ   = 0.d0
      VALJXXX = 0.d0   ! zero the PHOTOJ equivalents for wtd averaging
      SKPERDD = 0.d0
      SWMSQQ  = 0.d0
      OD18    = 0.d0
      OD18Q   = 0.d0
```